server/routes/public.js

```diff
@@ -28,6 +28,57 @@ router.get('/login', async (req, res) => {
     res.send(user);
 });
 
+<<<<<<< Updated upstream
+<<<<<<< HEAD
+=======
+>>>>>>> Stashed changes
+router.get('/api/create-user', async (req, res) => {
+    // joi verification
+    // Validate appropriate parameters are passed into the create account endpoint.
+    const paramSchema = Joi.object({
+       
+       //The following joi verifications make sure the values inputted adhere to the format requirements
+       //specified in the migrations> create-worker-table and migrations> create-student-table.
+
+        email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net', 'ca'] } }).max(320).required(),
+        password: Joi.string().min(3).max(40).required(),
+        phone: Joi.string().max(20).phoneNumber(),
+        firstName: Joi.string().min(1).max(20).required(),
+        lastName: Joi.string().min(1).max(20).required(),
+        studentId: Joi.number().positive().integer(),
+        workerId: Joi.number().positive().integer(),
+        userType: Joi.string().min(1).valid(['student', 'worker']),
+        type: Joi.string(),
+        specialization: Joi.string().valid(['social worker', 'guidance councellor']),//verify options
+        //access_token: Joi.string().alphanum().min(3).max(30).required(), 
+        //insert joi verification for first_name,last_name, type,student_id,workerID,phone
+    });
+    // the endpoint `/api/create-user` accepts ALL parameters for both user types, 
+    //as listed in the migrations> create-worker-table and migrations> create-student-table
+
+    const paramFirstName = req.query.firstName;
+    const paramLastName = req.query.lastName;
+    const paramType = req.query.type; //note: 'type' is called 'role' in the UI, but refers to either a worker or a student.
```

Comment:
Hi Nancy,
I recommend renaming this variable to `paramUserType` and the value to be `req.query.userType` to prevent repetition with line 67.

Reply:
Hi Amy,
Thank you for suggesting that!
My action: I've renamed this variable to `paramUserType` and the value to be `req.query.userType`.

You can now see these changes in line 67 of server/routes/public.js

Please confirm if this takes care of your suggestion!type: Joi.string(),